cg/cli/post_process/utils.py

```diff
@@ -0,0 +1,22 @@
+import re
+
+from cg.models.cg_config import CGConfig
+from cg.services.post_processing.pacbio.post_processing_service import PacBioPostProcessingService
+
+
+def get_post_processing_service_from_run_name(
+    context: CGConfig, run_name: str
+) -> PacBioPostProcessingService:
+    """
+    Get the correct post-processing service based on the run name.
+    Raises:
+        NameError if the run name is not recognized.
+    """
+    if _is_run_pacbio(run_name):
+        return context.pacbio_post_processing_service
+    raise NameError(f"Could not find a post-processing service for run name: {run_name}")
+
+
+def _is_run_pacbio(run_name: str) -> bool:
```

Comment:
with this maybe we can get rid of the name validation in the RunDataGenerator

Reply:
If we do it via pattern matching then we can make a pattern to service map. So we don't need to build up if statements.

Reply:
maybe we can implement this later when we have more post-processing types?

Reply:
implemented